components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt

```diff
@@ -0,0 +1,146 @@
+    }
+
+    // Sum of all tab's width.
+    val totalTabWidth = remember(tabWidths) {
+        tabWidths.sumOf { it.value.toInt() }.dp
+    }
+
+    Column(modifier = modifier) {
+        if (totalTabWidth > screenWidth) {
+            HmrcScrollableTabRow(tabItems, selectedIndex, onTabSelected, tabBarStyle)
+        } else {
+            HmrcTabRow(tabItems, selectedIndex, onTabSelected, tabBarStyle)
+        }
+    }
+}
+
+@Composable
+private fun HmrcTabRow(
+    tabItems: List<String>,
+    selectedIndex: Int,
+    onTabSelected: (Int) -> Unit,
+    tabBarStyle: TabBarViewStyle,
+) {
+    SecondaryTabRow(
+        selectedTabIndex = selectedIndex,
+        containerColor = tabBarStyle.backgroundColour,
+        divider = {}, // Empty divider, we don't want the grey divider at the bottom of the Tab.
+        indicator = { tabPositions ->
+            TabRowDefaults.SecondaryIndicator(
+                Modifier
+                    .tabIndicatorOffset(tabPositions[selectedIndex])
+                    .height(HmrcTheme.dimensions.hmrcSpacing4),
+                color = tabBarStyle.selectedTextColour
+            )
+        },
+        tabs = { HmrcTabItems(tabItems, selectedIndex, onTabSelected, tabBarStyle) }
+    )
+}
+
+@Composable
+private fun HmrcScrollableTabRow(
+    tabItems: List<String>,
+    selectedIndex: Int,
+    onTabSelected: (Int) -> Unit,
+    tabBarStyle: TabBarViewStyle,
+) {
+    SecondaryScrollableTabRow(
+        selectedTabIndex = selectedIndex,
+        containerColor = tabBarStyle.backgroundColour,
+        divider = {}, // Empty divider, we don't want the grey divider at the bottom of the Tab.
+        edgePadding = 0.dp,
+        indicator = { tabPositions ->
+            TabRowDefaults.SecondaryIndicator(
+                Modifier
+                    .tabIndicatorOffset(tabPositions[selectedIndex])
+                    .height(HmrcTheme.dimensions.hmrcSpacing4),
+                color = tabBarStyle.selectedTextColour
+            )
+        },
+        tabs = { HmrcTabItems(tabItems, selectedIndex, onTabSelected, tabBarStyle) }
+    )
+}
+
+@Composable
+private fun HmrcTabItems(
+    tabItems: List<String>,
+    selectedIndex: Int,
+    onTabSelected: (Int) -> Unit,
+    tabBarStyle: TabBarViewStyle,
+) {
+    run {
+        tabItems.forEachIndexed { index, tabText ->
+            Tab(
+                selected = index == selectedIndex,
+                onClick = { onTabSelected(index) },
+                selectedContentColor = tabBarStyle.selectedTextColour,
+                unselectedContentColor = HmrcGrey1
+            ) {
+                Text(
+                    text = tabText,
+                    style = HmrcTheme.typography.tabBarText,
+                    modifier = Modifier.padding(HmrcTheme.dimensions.hmrcSpacing16),
+                )
+            }
+        }
+    }
+}
+
+enum class TabBarViewStyle(val backgroundColour: Color, val selectedTextColour: Color) {
```

Comment:
Should these not be in with the rest of our Styles maybe?

Reply:
My thinking behind this is that these `TabBarViewStyle` only really gets used in this `TabBarView`. This really shouldn't be use anywhere else, hence why I've put it in this class directly. 

Is that ok, Or do you think this should still go somewhere else?